### PR TITLE
Use LinkIdentityRequested event instead of link_identity extrinsic

### PR DESCRIPTION
**Pull request by silva-fj**
This is the step to migrating to using events instead of extrinsics as indirect call triggers. I'll be adding one by one in separate PR's to make it easier to review

### CODE REVIEW

File: pallets/identity-management/src/lib.rs [hunk 1]
```diff
@@ -81,6 +81,10 @@ pub mod pallet {
 		// TODO: do we need account as event parameter? This needs to be decided by F/E
 		LinkIdentityRequested {
 			shard: ShardIdentifier,
+			account: T::AccountId,
```
Comment (silva-fj) [inline pallets/identity-management/src/lib.rs:84]: Do we need to update something on the clients/sdk for this ? @jonalvarezz

**Review by BillyWooo — COMMENTED** [review]
Over all, it looks good to me. 
I just recall my memory.  I think we had something like this before for Indirect Invocation mode. Then we removed it. Now we add it back. I agree to add it back. Just forget what's the reason to remove before.

File: pallets/identity-management/src/lib.rs (hunk 1 above)
Comment (jonalvarezz) [reply pallets/identity-management/src/lib.rs:84]: Thanks @silva-fj , I checked it up, and can confirm the type isn't use on the client-sdk. Furthermore, client-sdk doesn't not really on parachain other than for fetching metadata and registered enclave data.

**Comment by silva-fj** [comment]
> Over all, it looks good to me. I just recall my memory. I think we had something like this before for Indirect Invocation mode. Then we removed it. Now we add it back. I agree to add it back. Just forget what's the reason to remove before.

Maybe it's worth it to find the issue/PR that removed it to know the reason

**Comment by silva-fj** [comment]
I will create a new PR with new changes

**Pull request closed without merging**